packages/headless/src/features/advanced-search-queries/advanced-search-queries-state.ts

```diff
@@ -1,16 +1,53 @@
+export interface AdvancedSearchQueriesDefaultFiltersState {
+  /**
+   * The initial cq filter (e.g., `((q AND aq) OR dq) AND cq).
+   */
+  cq: string;
+
+  /**
+   * The initial aq filter (e.g., `((q AND aq) OR dq) AND cq).
+   */
+  aq: string;
+}
+
 export interface AdvancedSearchQueriesState {
   /**
    * The cq filter (e.g., `((q AND aq) OR dq) AND cq).
    */
   cq: string;
 
+  /**
+   * Whether the cq filter was manually set.
+   *
+   * If the cq was manually set and the cq is registered, the cq will not be overriden by the default filter.
+   */
+  cqWasSet: boolean;
```

Comment:
`cqWasSet` is important, since it's possible that `#restoreSearchParameters` may be called before `#registerAdvancedSearchQueries`, in which case we don't want to override it with the default.